### PR TITLE
fiano: init at 1.2.0

### PR DESCRIPTION
**Pull request by jmbaur**
###### Description of changes

Fiano is a suite of Go-based tools for modifying UEFI firmware.

This initial inclusion of fiano omits one package called `fittool`, since this
tool has a separate `modRoot` than the rest and its `go.sum` is not up to date,
so it does not build. I will be working with upstream to get this fixed.

I also added myself as a maintainer of this package, and thus adding myself to
the maintainer list as well.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See
      [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests)
    (look inside
    [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or
    [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in
    [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or
    [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are
    [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking)
    to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using
      `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all
      changes have to be committed, also see
      [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in
      `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or
        breaking
  - [ ] (Module updates) Added a release notes entry if the change is
        significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS
        module
- [x] Fits
      [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/fiano/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  vendorSha256 = "sha256-00ZSAVEmk2pNjv6fo++gnpIheK8lo4AVWf+ghXappnI=";
+
+  ldflags = [ "-s" "-w" ];
+
+  meta = with lib; {
+    homepage = "https://github.com/linuxboot/fiano";
+    description = "Go-based tools for modifying UEFI firmware";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ jmbaur ];
+    platforms = platforms.all;
```
Comment (natsukium) [inline pkgs/tools/misc/fiano/default.nix:34]:
```suggestion
```
`platforms.all` is the default for `buildGoModule`.

File: pkgs/tools/misc/fiano/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    "cmds/guid2english"
+    "cmds/microcode"
+    "cmds/utk"
+  ];
+
+  vendorSha256 = "sha256-00ZSAVEmk2pNjv6fo++gnpIheK8lo4AVWf+ghXappnI=";
+
+  ldflags = [ "-s" "-w" ];
+
+  meta = with lib; {
```
Comment (natsukium) [inline pkgs/tools/misc/fiano/default.nix:29]: It would be better to add meta.changelog.

Comment (jmbaur) [reply pkgs/tools/misc/fiano/default.nix:32]: added

File: maintainers/maintainer-list.nix
[the remark this answers is not in the document]
Comment (jmbaur) [reply maintainers/maintainer-list.nix:1]: it is now in two commits

File: pkgs/tools/misc/fiano/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+buildGoModule rec {
```
[the remark this answers is not in the document]
Comment (jmbaur) [reply pkgs/tools/misc/fiano/default.nix]: done

File: pkgs/tools/misc/fiano/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
+buildGoModule rec {
```
Comment (natsukium) [inline pkgs/tools/misc/fiano/default.nix:5]:
```suggestion
}:

buildGoModule rec {
```
Just a trivial point, but could you please add a blank line here?

**Review by natsukium — COMMENTED** [review]
It looks much better now.
It would be better to remove the "v" from "v1.2.0" in the commit message.

**Comment by jmbaur** [comment]
@natsukium thanks for the review! Let me know if there's anything else that looks off

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 232715` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fiano</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LTGM

**Comment by pbek** [comment]
1 package built:
fiano

Result of `nixpkgs-review pr 232715` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fiano</li>
  </ul>
</details>

**Review by pbek — APPROVED** [review]

File: pkgs/tools/misc/fiano/default.nix [hunk 5]
```diff
@@ -0,0 +1,40 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
+
+buildGoModule rec {
+  pname = "fiano";
+
+  version = "1.2.0";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/fiano/default.nix:9]:
```suggestion
  pname = "fiano";
  version = "1.2.0";
```

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**